static/js/narrow.js

```diff
@@ -339,10 +339,12 @@ exports.activate = function (raw_operators, opts) {
         }
     }
 
-    if (filter.contains_only_private_messages()) {
-        compose.update_closed_compose_buttons_for_private();
+    if (filter.has_operator('stream')) {
+        // Show the New Topic button in a stream narrows.
+        $(".compose_new_topic_button").attr("hidden", false);
```

Comment:
Minor nitpick, but having the implementation details of the compose module inside it is cleaner imo, even if it is just two 1 line functions. That way if we rename the button, we just have to change it in `compose.js` and not touch any code that wants to show/hide it.